scheduler/src/cook/prometheus_metrics.clj

```diff
@@ -0,0 +1,116 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+
+;; Declares prometheus metrics for cook scheduler.
+
+(ns cook.prometheus-metrics
+  (:require [iapetos.core :as prometheus]
+            [iapetos.export :as prometheus-export]))
+
+(def default-summary-quantiles {0.5 0.01 0.75 0.01 0.9 0.01 0.95 0.01 0.99 0.01 0.999 0.01})
+
+(defonce registry
+         ;; A global registry for all metrics reported by Cook.
+         ;; All metrics must be registered before they can be recorded.
+         ;; We are standardizing the metric format to be :cook/<component>-<metric-name>-<unit>
+         (-> (prometheus/collector-registry)
+           (prometheus/register
+             ;; Scheduler metrics --------------------------------------------------------------------------------------
+             ;; Note that we choose to use a summary instead of a histogram for the latency metrics because we only have
+             ;; one scheduler process running per cluster, so we do not need to aggregate data from multiple sources.
+             ;; The quantiles are specified as a map of quantile to error margin.
```

Comment:
Nit: would be helpful to duplicate this line up where the defaults are specified, but I ofund it pretty quickly.

Reply:
Ah good catch, I added the default value later and forgot this comment was here. I updated it.